version: '3.3'

services:
  mysql_db:
    image: linuxserver/mariadb:alpine
    restart: always
    environment:
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_USER: ${MYSQL_USER}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
    ports:
      - '3307:3306'
    volumes:
      - mysql_db:/var/lib/mysql
      - ./backend/database/init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      - my-network
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 10s
      timeout: 5s
      retries: 5
      start_period: 10s

  python_app:
    build: .
    restart: always
    depends_on:
      mysql_db:
        condition: service_healthy
    networks:
      - my-network

  flask_server:
    build:
      context: .
      dockerfile: Dockerfile.flask
    ports: 
      - '5000:5000'
    depends_on:
      mysql_db:
        condition: service_healthy
    networks:
      - my-network

volumes:
  mysql_db:

networks:
  my-network:
